File: samples/SampleCounterV4/Mvu/Widgets/Heading.cs
using Metano.TypeScript.DOM;

namespace SampleCounterV4.Mvu.Widgets;

public sealed class Heading(string content, int level = 1) : Widget
{
    public override HtmlElement Render()
    {
        var sizeEm = level switch
        {
            1 => 2.0,
            2 => 1.5,
            3 => 1.25,
            4 => 1.0,
            5 => 0.875,
            _ => 0.75,
        };

        var span = Js.Document.CreateElement(HtmlElementType.Span);
        span.TextContent = content;
        span.ApplyStyle($"display:block;font-weight:bold;font-size:{sizeEm}em");

        return span;
    }
}

File: targets/js/sample-counter-v4/src/mvu/widgets/heading.ts
import { Widget } from "#/mvu";

export class Heading extends Widget {
  private readonly _level: number;

  private readonly _content: string;

  constructor(content: string, level: number) {
    super();
    this._content = content;
    this._level = level;
  }

  render(): HTMLElement {
    const sizeEm = this._level === 1 ? 2 : this._level === 2 ? 1.5 : this._level === 3 ? 1.25 : this._level === 4 ? 1 : this._level === 5 ? 0.875 : 0.75;
    const span = document.createElement("span");
    span.textContent = this._content;
    span.setAttribute('style', `display:block;font-weight:bold;font-size:${sizeEm}em`);

    return span;
  }
}
